index.js

```diff
@@ -3586,6 +3593,53 @@
       [Fn(a, b), m(a), m(a)],
       Z.sortBy);
 
+  //# zipWith :: (a -> b -> c) -> Array a -> Array b -> Array c
+  //.
+  //. Returns the result of combining, pairwise, the given arrays using the
+  //. given binary function. The length of the resulting array is equal to the
+  //. length of the shorter input array.
+  //.
+  //. ```javascript
+  //. > S.zipWith(a => b => a + b, ['a', 'b'], ['x', 'y', 'z'])
+  //. ['ax', 'by']
+  //.
+  //. > S.zipWith(a => b => [a, b], [1, 3, 5], [2, 4])
+  //. [[1, 2], [3, 4]]
+  //. ```
+  function zipWith(f, xs, ys) {
+    var result = [];
+    var len = Math.min(xs.length, ys.length);
+    for (var idx = 0; idx < len; idx += 1) result.push(f(xs[idx])(ys[idx]));
+    return result;
+  }
+  S.zipWith =
+  def('zipWith',
+      {},
+      [Fn(a, Fn(b, c)), $.Array(a), $.Array(b), $.Array(c)],
+      zipWith);
+
+  //# zip :: Array a -> Array b -> Array (Pair a b)
+  //.
+  //. Returns the result of combining, pairwise, the given arrays into a
+  //. resulting array with length equal to the length of the shorter input
+  //. array.
```

Comment:
How about this?

_Returns an array of pairs of corresponding elements from the given arrays. The length of the resulting array is equal to the length of the shorter input array._

Please add “see also” links in both directions.

I wonder whether we should list `zip` before `zipWith`. What do you think?

Reply:
doin thy changes!